tests/utils/test_migration_helpers.py

```diff
@@ -251,25 +264,57 @@ def func_test_raise_unparametrized_decorator(param1: int) -> int:
 
 
 def test_default_function_deprecation_warning_texts_generic_version(
-    default_function_deprecation_texts_generic,
+    default_function_deprecation_texts_generic: str,
 ):
     assert "{function}" in default_function_deprecation_texts_generic
 
 
 def test_default_deprecation_warning_texts_specific_version(
-    default_deprecation_texts_specific: Callable,
+    default_param_deprecation_texts_specific: Callable,
 ):
     # accounts for different amounts of version parameters
     n_params = len(
-        inspect.signature(default_deprecation_texts_specific).parameters.keys()
+        inspect.signature(default_param_deprecation_texts_specific).parameters.keys()
     )
     versions = [f"1.0.{n}" for n in range(n_params)]
 
     @deprecate_parameters_if_set(
-        default_deprecation_texts_specific(*versions), "param1"
+        default_param_deprecation_texts_specific(*versions), "param1"
     )
     def func_test_raise_unparametrized_decorator(param1: int) -> int:
         return param1
 
     with pytest.warns(DeprecationWarning, match=".*" + ".*".join(versions) + ".*"):
         func_test_raise_unparametrized_decorator(1)
+
+
+@pytest.fixture(
+    scope="class", params=[False, True],
+)
+def is_test_optional_parameters(request) -> bool:
+    return request.param
+
+
+def test_get_specific_function_deprecation_warning(
+    is_test_optional_parameters: bool,
+    default_function_deprecation_texts_specific: Callable,
+):
+
+    if not is_test_optional_parameters:
+        default_function_deprecation_texts_specific(
+            function_name="test_func", deprecated_in="1.0.0"
+        )
+    else:
+        # accounts for different amounts of version parameters
+        n_params = (
+            len(
+                inspect.signature(
+                    default_param_deprecation_texts_specific
+                ).parameters.keys()
+            )
+            - 2
+        )
+        test_strings = [f"test_{n}" for n in range(n_params)]
+        default_function_deprecation_texts_specific(
+            function_name="test_func", deprecated_in="1.0.0", *test_strings
+        )
```

Comment:
We don't assert anything in these tests, right? Does it happen within `default_function_deprecation_texts_specific` callable? Or some side effect is triggered here, i.e., warning?

Reply:
I just wanted to assure that future function deprecation warnings contain a function_name and a deprecated_in parameter at least.
I would also make a assertion on the argspec if that suits you better.

Reply:
Ah, I see. Yeah, I mean, the test is a bit too broad. If my understanding is correct, `get_specific_function_deprecation_warning` returns a string. So the test creates a string but doesn't compare it with anything, and doesn't print anything either, right? So the statement can return anything and the test still succeed.

Reply:
- [ ] Add a comment and type-check on the result

Reply:
Solved with latest commits.